src/DebugPoints/DebugPointManager.class.st

```diff
@@ -0,0 +1,374 @@
+		  accessStrategy: anAccessStrategySymbol
+		  withBehaviors: {  }
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableNamed: aSlotNameSymbol accessStrategy: anAccessStrategySymbol withBehaviors: aListOfBehaviorClasses [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	| dp instanceVariable |
+	instanceVariable := aClass slotNamed: aSlotNameSymbol.
+	dp := aDebugPointClass new
+		      instanceVariable: instanceVariable
+		      accessStrategy: anAccessStrategySymbol;
+		      yourself.
+
+	^ self installDebugPoint: dp withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableReadNamed: aSlotNameSymbol [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #read
+		  withBehaviors: {  }
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableReadNamed: aSlotNameSymbol withBehaviors: aListOfBehaviorClasses [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #read
+		  withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableWriteNamed: aSlotNameSymbol [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #write
+		  withBehaviors: {  }
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableWriteNamed: aSlotNameSymbol withBehaviors: aListOfBehaviorClasses [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #write
+		  withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'announcements' }
+DebugPointManager class >> notifyDebugPointAdded: aDebugPoint [
+
+	| announcement |
+	announcement := DebugPointAdded 
+		on: aDebugPoint
+		nodes: aDebugPoint link nodes.
```

Comment:
On the question of high/low level concernes, I wonder if the nodes should be known by the model or only by the underlying implementation? @MarcusDenker do you have an opinion on that?